R/comparePlot.R

```diff
@@ -335,55 +414,49 @@ comparePlot <- function(compareDF,
-#' @import magrittr
+#' @importFrom dplyr filter arrange bind_cols rename_with
+#' @importFrom magrittr set_rownames
 #'
 #' @export
 comparePrep <- function(contrastList,
                         valueCol = "logFC",
                         significanceCol = "P.Value"){
-
     assertthat::assert_that(length(contrastList) == 2,
-                            !is.null(names(contrastList)),
-                            "data.frame" %in% class(contrastList[[1]]),
-                            "data.frame" %in% class(contrastList[[2]]),
-                            msg = "contrastList must be a named list of length 2 where both items are of class 'data.frame'.")
+                           !is.null(names(contrastList)),
+                           "data.frame" %in% class(contrastList[[1]]),
+                           "data.frame" %in% class(contrastList[[2]]),
+                           msg = "contrastList must be a named list of length 2 where both items are of class 'data.frame'.")
     assertthat::assert_that(valueCol %in% colnames(contrastList[[1]]),
-                            valueCol %in% colnames(contrastList[[2]]),
-                            msg = "The valueCol must be included in the colnames of both items of contrastList.")
+                           valueCol %in% colnames(contrastList[[2]]),
+                           msg = "The valueCol must be included in the colnames of both items of contrastList.")
     assertthat::assert_that(significanceCol %in% colnames(contrastList[[1]]),
-                            significanceCol %in% colnames(contrastList[[2]]),
-                            msg = "The significanceCol must be included in the colnames of both items of contrastList.")
-
-    ttNames <- names(contrastList)
-    tt1 <- contrastList[[1]]
-    tt2 <- contrastList[[2]]
+                           significanceCol %in% colnames(contrastList[[2]]),
+                           msg = "The significanceCol must be included in the colnames of both items of contrastList.")
 
-    commonIDs <- intersect(rownames(tt1), rownames(tt2))
-    assertthat::assert_that(!length(commonIDs) < 1,
-                            msg = "No common gene IDs were found between the two dataframes in contrastList.")
+    commonIDs <- intersect(rownames(contrastList[[1]]), rownames(contrastList[[2]]))
+    assertthat::assert_that(length(commonIDs) > 0,
+                           msg = "No common gene IDs were found between the two dataframes in contrastList.")
 
     # Filter both tables to the same set of genes in the same order
-    tt1 <- data.frame("geneid" = row.names(tt1), tt1, row.names = NULL)
-    tt2 <- data.frame("geneid" = row.names(tt2), tt2, row.names = NULL)
-    tt1 <- tt1 %>%
-        dplyr::filter(geneid %in% commonIDs) %>%
-        dplyr::arrange(geneid)
-    tt2 <- tt2 %>%
-        dplyr::filter(geneid %in% commonIDs) %>%
-        dplyr::arrange(geneid)
-
+    tt1 <- construct_compartor_data(contrastList[[1]], commonIDs)
```

Comment:
Typo in the word compartor -> It needs to be comparator

Reply:
done